src/cpp/rtps/security/SecurityManager.cpp

```diff
@@ -143,7 +143,7 @@ bool SecurityManager::init(
         log_options.log_level = LoggingLevel::ERROR_LEVEL;
         log_options.log_file = "";
 
-        const auto init_logging_fail = [&](){
+        const auto init_logging_fail = [this](SecurityException& exception){
```

Comment:
Why capture `this` as opposed to simply the logging plugin?

Reply:
`logging_plugin_` is a *member variable* of `SecurityManager` (`this`) thus cannot be captured by itself. It is treated as `(*this).logging_plugin_` and `*this` is always implicitly captured by reference. Therefore I'm only making the implicit capture explicit.

Reply:
No problem, although note that cpp14 should be fine (they're talking about making cpp17 the standard for Foxy anyway).